src/resource/solidDataset.ts

```diff
@@ -1156,14 +1156,41 @@ function resolveLocalIrisInThing(
 }
 
 /**
- * Fetch the contents of '.well-known/solid' for a given resource URL.
+ * Fetch a SolidDataset containing information about the capabilities of the
+ * storage server that hosts the given resource URL. For more information,
+ * please see the [ESS
+ * Documentation](https://docs.inrupt.com/ess/latest/services/discovery-endpoint/#well-known-solid).
+ *
+ * **Note:** The data contained in this dataset has changed between ESS 1.1 and
+ * ESS 2.0, as such you will need to check for multiple predicates to support
+ * both versions.
+ *
+ * ```typescript
+ * const wellKnown = await getWellKnownSolid(resource);
+ *
+ * // The wellKnown dataset uses a blank node for the subject all of it’s predicates,
+ * // such that we need to call getThingAll with acceptBlankNodes set to true to
+ * // retrieve back predicates contained within the dataset
+ * const wellKnownSubjects = getThingAll(wellKnown, {
+ *   acceptBlankNodes: true,
+ * });
+ * const wellKnownSubject = wellKnownSubjects[0];
+ *
+ * // Retrieve a value from the wellKnown dataset:
+ * let notificationGateway = getIri(
+ *   wellKnownSubject,
+ *   "http://www.w3.org/ns/solid/terms#notificationGateway"
+ * );
```

Comment:
So what we _should_ be doing is going through the dataset and look for all quads in the default graph with `http://www.w3.org/ns/solid/terms#notificationGateway` as predicate, and hope there is only one.

I am not sure whether the `Thing` API supports it, but I fear it does not. Does `SolidDataset` somehow support this? https://docs.inrupt.com/developer-tools/api/javascript/solid-client/modules/resource_solidDataset.html#getsoliddataset is unclear about that.

If it does not, then we need to loop over all things, and over each of their predicates.

And we should stop depending on Solid Thing because it does not make sense to approach RDF this way, especially not at the low level we are parsing at here.

Reply:
Nah it does not.

Reply:
I don't believe there is any way to iterate over all predicates using the Thing APIs.. I guess we could write:

```
let notificationGateway = wellKnownSubjects.reduce((subject) => {
  const modern = getIri(subject, "http://www.w3.org/ns/solid/terms#notificationGateway");
  const legacy = getIri(subject, "http://inrupt.com/ns/ess#notificationGatewayEndpoint");

  if (modern) {
    return modern;
  }

  if (legacy) {
    return legacy;
  }

  return null;
}, null);
```

Or something..

Reply:
But is that really better?

Reply:
Nah, and it's a bug in the Thing design.